manifests/config/resource_source.pp

```diff
@@ -64,48 +64,42 @@
 # }
 #
 define rundeck::config::resource_source(
-  $directory                          = $rundeck::params::default_resource_dir,
-  $include_server_node                = $rundeck::params::include_server_node,
-  $mapping_params                     = '',
-  $number                             = '1',
-  $project_name                       = undef,
-  $resource_format                    = $rundeck::params::resource_format,
-  $running_only                       = true,
-  $script_args                        = '',
-  $script_args_quoted                 = $rundeck::params::script_args_quoted,
-  $script_file                        = '',
-  $script_interpreter                 = $rundeck::params::script_interpreter,
-  $source_type                        = $rundeck::params::default_source_type,
-  $url                                = '',
-  $url_cache                          = $rundeck::params::url_cache,
-  $url_timeout                        = $rundeck::params::url_timeout,
-  $use_default_mapping                = true,
-  $puppet_enterprise_host             = '',
-  $puppet_enterprise_port             = '',
-  $puppet_enterprise_ssl_dir          = '',
-  $puppet_enterprise_mapping_file     = '',
-  $puppet_enterprise_metrics_interval = '',
+  Stdlib::Absolutepath $directory                                = $rundeck::params::default_resource_dir,
+  Boolean $include_server_node                                   = $rundeck::params::include_server_node,
+  $mapping_params                                                = '',
+  Integer $number                                                = 1,
+  Optional[String] $project_name                                 = undef,
+  Enum['resourcexml', 'resourceyaml'] $resource_format           = $rundeck::params::resource_format,
+  Boolean $running_only                                          = true,
+  String $script_args                                            = '',
+  Boolean $script_args_quoted                                    = $rundeck::params::script_args_quoted,
+  Optional[Stdlib::Absolutepath] $script_file                    = undef,
+  String $script_interpreter                                     = $rundeck::params::script_interpreter,
+  Rundeck::Sourcetype $source_type                               = $rundeck::params::default_source_type,
+  String $url                                                    = '',
```

Comment:
Optional[Stdlib::Httpurl] or not?

Reply:
@juniorsysadmin I think I didn't do this because it's currently set to be an empty string. I only did the bare minimum of converting things that were not undef to undefs to get rid of `validate_xx` routines. But I think we should switch this to `Optional[Stdlib::Httpurl]` in another PR if / when we get rid of remaining empty strings in favor of `undef`